### PR TITLE
Patched cryptonite-0.26

### PR DESCRIPTION
**Pull request by jneira**
I had to force it use a previous version of memory cause `memory-0.14.18` has been hard to patch. 

I hope ci builds returns to green with this

### CODE REVIEW

**Comment by jneira** [comment]
The ci build fails now due to amazonka-core-1.6.0 -> typed-process-0.2.6.0:
```
Failed to build typed-process-0.2.6.0.
Build log (
/root/.etlas/logs/eta-0.8.6.5/typed-process-0.2.6.0-333a0c2f6c45c602801f734ef0293a364cb3ec21378ad0e82c92620846ed8181.log
):
Configuring library for typed-process-0.2.6.0..
Preprocessing library for typed-process-0.2.6.0..
Building library for typed-process-0.2.6.0..

[3 of 3] Compiling System.Process.Typed
src/System/Process/Typed.hs:116:23: error:
    Module ‘GHC.RTS.Flags’ does not export ‘getConcFlags’
    |
116 | import GHC.RTS.Flags (getConcFlags, ctxtSwitchTime)
    |                       ^^^^^^^^^^^^

src/System/Process/Typed.hs:116:37: error:
    Module ‘GHC.RTS.Flags’ does not export ‘ctxtSwitchTime’
    |
116 | import GHC.RTS.Flags (getConcFlags, ctxtSwitchTime)
    |                                     ^^^^^^^^^^^^^^

src/System/Process/Typed.hs:120:28: error:
    Module ‘System.Posix.Types’ does not export ‘GroupID’
    |
120 | import System.Posix.Types (GroupID, UserID)
    |                            ^^^^^^^

src/System/Process/Typed.hs:120:37: error:
    Module ‘System.Posix.Types’ does not export ‘UserID’
    |
120 | import System.Posix.Types (GroupID, UserID)
    |                                     ^^^^^^
etlas: Failed to build typed-process-0.2.6.0 (which is required by
amazonka-core-1.6.0). See the build log above for details.
```

File: patches/cryptonite-0.26.patch [hunk 1]
```diff
@@ -0,0 +1,3536 @@
...
+-    hashInternalFinalize      = c_md4_finalize
+-
+-foreign import ccall unsafe "cryptonite_md4_init"
+-    c_md4_init :: Ptr (Context a)-> IO ()
+-
+-foreign import ccall "cryptonite_md4_update"
+-    c_md4_update :: Ptr (Context a) -> Ptr Word8 -> Word32 -> IO ()
+-
+-foreign import ccall unsafe "cryptonite_md4_finalize"
+-    c_md4_finalize :: Ptr (Context
```
Comment (jneira) [inline patches/cryptonite-0.26.patch]: Not sure if we should comment this, but extract the `Context` from the `Ptr` supposes some yak-shaving

**Comment by jneira** [comment]
@rahulmutt i think we could merge this though

**Comment by rahulmutt** [comment]
Merging for now, let's take care of the CI stuff later. Thanks @jneria!

**Review by rahulmutt — APPROVED** [review]

**Pull request merged**